vsphere_plugin_common/__init__.py

```diff
@@ -292,37 +482,60 @@ def create_server(self,
                       vm_name,
                       os_type='linux',
                       domain=None,
-                      dns_servers=None):
+                      dns_servers=None,
+                      allowed_hosts=None,
+                      allowed_clusters=None,
+                      allowed_datastores=None):
         ctx.logger.debug("Entering create_server with parameters %s"
                          % prepare_for_log(locals()))
-        host, datastore = self.place_vm(auto_placement)
 
-        # This should be debug, but left as info until CFY-4867 makes logs
-        # more visible
-        ctx.logger.info("Using datastore %s for manager node." % datastore)
+        self._validate_inputs(
+            allowed_hosts=allowed_hosts,
+            allowed_clusters=allowed_clusters,
+            allowed_datastores=allowed_datastores,
+            template_name=template_name,
+            networks=networks,
+            resource_pool_name=resource_pool_name,
+            datacenter_name=datacenter_name,
+            vm_cpus=cpus,
+            vm_memory=memory,
+        )
+
+        candidate_hosts = self.find_candidate_hosts(
+            resource_pool=resource_pool_name,
+            vm_cpus=cpus,
+            vm_memory=memory,
+            vm_networks=networks,
+            allowed_hosts=allowed_hosts,
+            allowed_clusters=allowed_clusters,
```

Comment:
Same reasoning as on line 501- avoiding multiple lines being changed when one line is 'really' changed.